### ISSUE TITLE
Regression

### ISSUE THREAD

**Issue by zoeyzhougs** [goldmansachs/gs-quant#94]
**Describe the problem.**
Regression should be available in gs-quant and PlotToolPro.

**Describe the solution you'd like**
Add a class in gs-quant (constructor in PlotToolPro) for the model fitting which has methods to get model parameters and perform prediction. 
For example, linear regression takes the form `LinearRegression(X, y, w)` where X denotes explanatory variables [x1, x2, x3, ...], y is dependent variable and w is moving window. It has methods:
`coefficient(i)` - the i-th coefficient
`rsquare()` - model R squared
`tstat(i)` - t-stat of the i-th coefficient
`predict(X')` where X' is test set of explanatory variables - predicted y

**Are you willing to contribute**
Yes

**Comment by francisg77** [comment]
Looks good to me, should we do _r_square_ and _t_stat_ just to be consistent with other methods.

what does an example with _predict_ look like?

**Comment by zoeyzhougs** [comment]
Agreed on the namings. 
For now it's hard to use the predict function as we don't have functions to easily split data into train and test subsets. One example that needs improve: 
> spx = returns(SPX.spot(), simple)
> spx_vol = diff(SPX.implied_volatility(1m, forward, 100), 1)
> spx_train = interpolate(spx, [2020-02-01, 2020-02-02, 2020-02-03])
> spx_vol_train = interpolate(spx_vol, [2020-02-01, 2020-02-02, 2020-02-03])
> spx_test = interpolate(spx, [2020-02-04, 2020-02-05])
> r = LinearRegression([spx_train], spx_vol_train)
> r.predict(spx_test)

**Comment by zoeyzhougs** [comment]
renaming `r_square` to `r_squared`

**Comment by Atharva-Peshkar** [comment]
> Agreed on the namings.
> For now it's hard to use the predict function as we don't have functions to easily split data into train and test subsets. One example that needs improve:
> 
> > spx = returns(SPX.spot(), simple)
> > spx_vol = diff(SPX.implied_volatility(1m, forward, 100), 1)
> > spx_train = interpolate(spx, [2020-02-01, 2020-02-02, 2020-02-03])
> > spx_vol_train = interpolate(spx_vol, [2020-02-01, 2020-02-02, 2020-02-03])
> > spx_test = interpolate(spx, [2020-02-04, 2020-02-05])
> > r = LinearRegression([spx_train], spx_vol_train)
> > r.predict(spx_test)

I'm a student and have been working in ML for a while now. I'm a new contributor here. We could use the train_test_split function from sklearn library for easily splitting the training and test datasets. It'll help with the predict function. I would've done it but am not able to find PlotToolPro.

**Comment by andyphillipsgs** [comment]
@Atharva-Peshkar thanks for the input. we can run an example which is pure python, you don't need access to the [PlotTool Pro](https://developer.gs.com/discover/plottool/) application as the functions being discussed are exposed in the timeseries module in GS Quant. Give us a shout if you are keen to extend and we can give some guidance and add a tutorial

**Comment by Atharva-Peshkar** [comment]
@andyphillipsgs  Yeah! sure I'd be glad to contribute.

> @Atharva-Peshkar thanks for the input. we can run an example which is pure python, you don't need access to the [PlotTool Pro](https://developer.gs.com/discover/plottool/) application as the functions being discussed are exposed in the timeseries module in GS Quant. Give us a shout if you are keen to extend and we can give some guidance and add a tutorial

**Comment by sirinath** [comment]
It might be an idea to open source plot tools.
